File: CS-Examples/09_Charts/AddTrendline.vb
Imports Spire.Xls

Namespace AddTrendline

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object to represent an Excel workbook
            Dim workbook As New Workbook()

            ' Load an existing Excel file named "ChartSample2.xlsx" from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ChartSample2.xlsx")

            ' Get the first worksheet (index 0) from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Get the first chart (index 0) on the worksheet
            Dim chart As Chart = sheet.Charts(0)

            ' Set the chart title to "Logarithmic Trendline"
            chart.ChartTitle = "Logarithmic Trendline"

            ' Add a logarithmic trendline to the first series of the chart
            chart.Series(0).TrendLines.Add(TrendLineType.Logarithmic)

            ' Get the second chart (index 1) on the worksheet
            Dim chart1 As Chart = sheet.Charts(1)

            ' Set the chart title to "Moving Average Trendline"
            chart1.ChartTitle = "Moving Average Trendline"

            ' Add a moving average trendline to the first series of the chart
            chart1.Series(0).TrendLines.Add(TrendLineType.Moving_Average)

            ' Get the third chart (index 2) on the worksheet
            Dim chart2 As Chart = sheet.Charts(2)

            ' Set the chart title to "Linear Trendline"
            chart2.ChartTitle = "Linear Trendline"

            ' Add a linear trendline to the first series of the chart
            chart2.Series(0).TrendLines.Add(TrendLineType.Linear)

            ' Get the fourth chart (index 3) on the worksheet
            Dim chart3 As Chart = sheet.Charts(3)

            ' Set the chart title to "Exponential Trendline"
            chart3.ChartTitle = "Exponential Trendline"

            ' Add an exponential trendline to the first series of the chart
            chart3.Series(0).TrendLines.Add(TrendLineType.Exponential)

            ' Specify the output filename for the modified workbook
            Dim output As String = "AddTrendline.xlsx"

            ' Save the modified workbook to a new Excel file with the Excel 2013 format
            workbook.SaveToFile(output, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the Excel file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace

File: CS-Examples/09_Charts/AddTrendline.cs
using System;
using System.Windows.Forms;
using Spire.Xls;

namespace AddTrendline
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            //Create a workbook
			Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ChartSample2.xlsx");

            // Get the first worksheet 
            Worksheet sheet = workbook.Worksheets[0];

            //select chart and set logarithmic trendline
            Chart chart = sheet.Charts[0];
            chart.ChartTitle = "Logarithmic Trendline";
            chart.Series[0].TrendLines.Add(TrendLineType.Logarithmic);

            //select chart and set moving_average trendline
            Chart chart1 = sheet.Charts[1];
            chart1.ChartTitle = "Moving Average Trendline";
            chart1.Series[0].TrendLines.Add(TrendLineType.Moving_Average);

            //select chart and set linear trendline
            Chart chart2 = sheet.Charts[2];
            chart2.ChartTitle = "Linear Trendline";
            chart2.Series[0].TrendLines.Add(TrendLineType.Linear);

            //select chart and set exponential trendline
            Chart chart3 = sheet.Charts[3];
            chart3.ChartTitle = "Exponential Trendline";
            chart3.Series[0].TrendLines.Add(TrendLineType.Exponential);

            //Save the document
            string output = "AddTrendline.xlsx";
			workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the Excel file
            ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}
